nsot/api/views.py

```diff
@@ -424,6 +424,23 @@ class NetworkViewSet(ResourceViewSet):
     lookup_value_regex = '[a-fA-F0-9:.]+(?:\/\d+)?'
     natural_key = 'cidr'
 
+    def allocate_networks(self, networks, prefix_length, site_pk, parent, state='allocated'):
+        site = models.Site.objects.get(pk=site_pk)
+        for n in networks:
+            obj = models.Network(
```

Comment:
I think it would be cleaner to remove the `prefix_length` argument from this method and just allow it to take a list of CIDRs.

Because broadcast_address, ip_version, parent are all automatically populated and overridden internally so providing them up front is redundant.

Then this could just be:

```python
for n in networks:
    obj = models.Network.create(cidr=n, site=site, state=state)
```